docs/modules/SPEC.md

```diff
@@ -0,0 +1,80 @@
+# Module Specification
+
+This document attempts to outline the recommended structure of Cosmos SDK modules.
+However, the ideas outlined here are meant to be applied as suggestions. Application
+developers are encouraged to improve upon and contribute to module structure and
+development design.
+
+## Structure
+
+A typical Cosmos SDK module can be structured as follows:
+
+```shell
+x/{module}
+├── abci.go
+├── alias.go
+├── client
+│   ├── cli
+│   │   ├── query.go
+│   │   └── tx.go
+│   └── rest
+│       ├── query.go
+│       └── tx.go
+├── exported
+│   └── exported.go
+├── genesis.go
+├── handler.go
+├── internal
+│   ├── keeper
+│   │   ├── invariants.go
+│   │   ├── keeper.go
+│   │   ├── ...
+│   │   └── querier.go
+│   └── types
+│       ├── codec.go
+│       ├── errors.go
+│       ├── events.go
+│       ├── expected_keepers.go
+│       ├── keys.go
```

Comment:
Maybe `keys` require its own description as well to standardize the use of prefix bytes for the store

Reply:
Can you elaborate on "standardize the use of prefix bytes"? Developers are allowed to use whatever they wish for prefixes.